clippy_lints/src/lib.rs

```diff
@@ -443,6 +447,50 @@ pub fn register_pre_expansion_lints(store: &mut rustc_lint::LintStore, sess: &Se
     store.register_pre_expansion_pass(move || Box::new(attrs::EarlyAttributes { msrv }));
 }
 
+fn read_msrv(conf: &Conf, sess: &Session) -> Option<RustcVersion> {
+    let clippy_msrv = conf.msrv.as_ref().and_then(|s| {
+        parse_msrv(s, None, None).or_else(|| {
+            sess.err(&format!(
+                "error reading Clippy's configuration file. `{}` is not a valid Rust version",
+                s
+            ));
+            None
+        })
+    });
+
+    let cargo_msrv = if_chain! {
+        if let Ok(manifest_dir) = std::env::var("CARGO_MANIFEST_DIR");
+        if let Ok(mut file) = File::open(format!("{manifest_dir}/Cargo.toml"));
+        let mut cargo_content_str = String::new();
+        if let Ok(_) = file.read_to_string(&mut cargo_content_str);
+        if let Ok(cargo_content) = toml::from_str::<Value>(&cargo_content_str);
+        if let Some(package) = cargo_content.get("package");
+        if let Some(rust_version) = package.get("rust-version");
+        if let Some(rust_version_str) = rust_version.as_str();
+        then {
+            parse_msrv(rust_version_str, None, None)
+        } else {
+            None
+        }
+    };
+
+    if let Some(cargo_msrv) = cargo_msrv {
+        if let Some(clippy_msrv) = clippy_msrv {
+            // if both files have an msrv, let's compare them and emit a warning if they differ
+            if clippy_msrv != cargo_msrv {
+                sess.warn(&format!(
+                    "the MSRV in `clippy.toml` and `Cargo.toml` differ; using `{}`",
+                    clippy_msrv
+                ));
+                return Some(clippy_msrv);
+            }
```

Comment:
```suggestion
            }
            return Some(clippy_msrv);
```

I think the code is a bit more readable if the `return` is moved out of the error case.

Reply:
I noticed, that I can omit the `return` entirely, thanks to your suggestion.

Makes it somewhat more readable, you're right